src/main/java/carpet/patches/NetHandlerPlayServerFake.java

```diff
@@ -27,6 +30,26 @@ public void disconnect(Component message)
             ((EntityPlayerMPFake) player).kill(message);
         }
     }
+    private boolean hasSpawned = false;
+
+    @Override
+    public void teleport(double d, double e, double f, float g, float h, Set<RelativeMovement> set)
+    {
+        super.teleport(d, e, f, g, h, set);
+
+        handleAcceptTeleportPacket(
+            new ServerboundAcceptTeleportationPacket(this.awaitingTeleport)
+        );
+
+        if (!hasSpawned) {
+            hasSpawned = true;
+        } else {
+            handleMovePlayer(
```

Comment:
what stops you from calling that every time (for simplicity sake)?

Reply:
Yeah this is a kludge; the spawn command fails though, I didn't look into it properly (hard slog, had figured the bug out, got lazy with the loose end), I'll take a another look and see if I can figure it out.

Reply:
Didn't figure out exactly what was happening here, but the teleport() indirectly called in createFake() is not playing nice, perhaps because things aren't entirely constructed and we start using the instance as if it were ('handling' packets for it).
Also suspect getting rid of it may mean special casing somewhere else.